students/youjin/westagram/users/views.py

```diff
@@ -1,3 +1,41 @@
-from django.shortcuts import render
+from django.http  import JsonResponse
+from django.views import View
 
-# Create your views here.
+from .models import User
+
+import json
+import re
+
+class UserView(View):
+    def post(self, request):
+        # users_email    = re.compile()
+        # users_password = re.compile()
+        data              = json.loads(request.body)
+        user_name         = data['name']
+        user_email        = data['email']
+        user_password     = data['password']
+        user_phone_number = data['phone_number']
+        user_gender       = data['gender']
+        user_sns          = data['sns']
+
+        try:
+            if re.match('^([a-z0-9_\.-]+)@([\da-z\.-]+)\.([a-z\.]{2,6})$', user_email) is None :  
+                return JsonResponse({"message": "KEY_ERROR"}, status=400)
+            
+            if User.objects.filter(email=user_email).exists():
+                return JsonResponse({"message" : "이미 존재하는 이메일입니다. 이메일을 다시 입력해주세요."}, status=400)
+            
+            if re.match('^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9]).{8,}$', user_password) is None:
+                return JsonResponse({"message": "KEY_ERROR"}, status=400)
```

Comment:
파이썬에서는 if 조건문 평가시, 반환 되는 값 자체가 boolean 일 경우나, Truthy, Falsy 한 값일 경우,  == False == None == 0등으로 평가하지 않고, 그 값을 그대로 사용합니다. 그 자체로도 True or False를 판단할 수 있기 때문입니다. False한 지 평가하실 때는 not 을 활용해보세요 :)
```suggestion
            if not re.match('^([a-z0-9_\.-]+)@([\da-z\.-]+)\.([a-z\.]{2,6})$', user_email):  
                return JsonResponse({"message": "KEY_ERROR"}, status=400)
            
            if not re.match('^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9]).{8,}$', user_password):
                return JsonResponse({"message": "KEY_ERROR"}, status=400)

            if User.objects.filter(email=user_email).exists():
                return JsonResponse({"message" : "email_already_exists"}, status=400)                
```